receiver/githubactionsreceiver/receiver_test.go

```diff
@@ -108,6 +109,48 @@ func TestEventToTracesTraces(t *testing.T) {
 	}
 }
 
+func TestEventToMetrics(t *testing.T) {
+	tests := []struct {
+		desc            string
+		payloadFilePath string
+		eventType       string
+		expectedMetrics int
+	}{
+		{
+			desc:            "WorkflowJobEvent processing",
+			payloadFilePath: "./testdata/queued/1_workflow_job_queued.json",
+			eventType:       "workflow_job",
+			expectedMetrics: 1,
+		},
+	}
+
+	logger := zaptest.NewLogger(t)
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			payload, err := os.ReadFile(test.payloadFilePath)
+			require.NoError(t, err)
+
+			event, err := github.ParseWebHook(test.eventType, payload)
+			require.NoError(t, err)
+
+			mh := newMetricsHandler(receivertest.NewNopSettings(), &Config{
+				MetricsBuilderConfig: metadata.MetricsBuilderConfig{
+					Metrics: metadata.MetricsConfig{
+						WorkflowJobsTotal: metadata.MetricConfig{
+							Enabled: true,
+						},
+					},
+				},
+			}, logger.Named("metricsHandler"))
+
+			metrics := mh.eventToMetrics(event.(*github.WorkflowJobEvent))
+
+			require.Equal(t, test.expectedMetrics, metrics.MetricCount(), fmt.Sprintf("%s: unexpected number of metrics", test.desc))
+			require.Equal(t, len(metadata.MapAttributeCiGithubWorkflowJobStatus), metrics.DataPointCount(), fmt.Sprintf("%s: unexpected number of datapoints", test.desc))
```

Comment:
nit: You could also use `require.Len` here 🙂

Reply:
`require.Len` asserts on the length of that map, not on `metrics.DataPointCount()`. While the test would fail correctly, we want to assert that `metrics.DataPointCount()` has the same amount of datapoints as the items in that map, not the other way around. 

by using `expect.Len` a failure looks something like `"metadata.MapAttributeCiGithubWorkflowJobStatus" should have X item(s), but has Y`, which is misleading as the expectation is the opposite.